editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -819,6 +825,9 @@ void VisualShaderEditor::edit(VisualShader *p_visual_shader) {
 		}
 		visual_shader = Ref<VisualShader>(p_visual_shader);
 		graph_plugin->register_shader(visual_shader.ptr());
+		if (!visual_shader->is_connected("changed", callable_mp(this, &VisualShaderEditor::_on_shader_changed))) {
+			visual_shader->connect("changed", callable_mp(this, &VisualShaderEditor::_on_shader_changed));
+		}
 		if (!visual_shader->is_connected("changed", callable_mp(this, &VisualShaderEditor::_update_preview))) {
```

Comment:
Sounds like you could remove the connection to `_update_preview` and call it in `_on_shader_changed` instead?

Reply:
I can combine them instead. Otherwise, there is no guarantee the previews will be updated.